presto-verifier/src/main/java/com/facebook/presto/verifier/checksum/ArrayColumnValidator.java

```diff
@@ -16,36 +16,87 @@
 import com.facebook.presto.common.type.ArrayType;
 import com.facebook.presto.common.type.RowType;
 import com.facebook.presto.common.type.Type;
+import com.facebook.presto.sql.tree.ArithmeticUnaryExpression;
+import com.facebook.presto.sql.tree.Cast;
 import com.facebook.presto.sql.tree.CoalesceExpression;
+import com.facebook.presto.sql.tree.ComparisonExpression;
 import com.facebook.presto.sql.tree.Expression;
-import com.facebook.presto.sql.tree.FunctionCall;
+import com.facebook.presto.sql.tree.Identifier;
+import com.facebook.presto.sql.tree.LambdaArgumentDeclaration;
+import com.facebook.presto.sql.tree.LambdaExpression;
 import com.facebook.presto.sql.tree.LongLiteral;
-import com.facebook.presto.sql.tree.QualifiedName;
 import com.facebook.presto.sql.tree.SingleColumn;
 import com.facebook.presto.sql.tree.TryExpression;
 import com.facebook.presto.verifier.framework.Column;
+import com.facebook.presto.verifier.framework.VerifierConfig;
 import com.google.common.collect.ImmutableList;
 
+import javax.inject.Inject;
+
 import java.util.List;
 import java.util.Objects;
 import java.util.Optional;
 
+import static com.facebook.presto.common.type.DoubleType.DOUBLE;
 import static com.facebook.presto.sql.QueryUtil.functionCall;
+import static com.facebook.presto.sql.QueryUtil.identifier;
 import static com.facebook.presto.verifier.framework.VerifierUtil.delimitedIdentifier;
 import static com.google.common.base.Preconditions.checkArgument;
 
 public class ArrayColumnValidator
         implements ColumnValidator
 {
+    private final double relativeErrorMargin;
+    private final double absoluteErrorMargin;
+    private final boolean arrayFloatingPointErrorMargin;
+
+    @Inject
+    public ArrayColumnValidator(VerifierConfig config)
+    {
+        this.relativeErrorMargin = config.getRelativeErrorMargin();
```

Comment:
I wonder if we should inject FloatingPointColumnValidator instead, and then can just call floatingPointColumnValidator.validate() and leave these configs as implementation details of the floatingpointcolumnvalidator.

Reply:
@rschlussel 
Do you mean make FloatingPointColumnValidator a member of ArrayColumnValidator and pass the former to the constructor of the latter?
We can do this.